src/main/kotlin/com/github/hanseter/json/editor/SchemaNormalizer.kt

```diff
@@ -69,12 +69,25 @@ object SchemaNormalizer {
                     resolutionScope, schema
             )
         } else {
-            resolveRefs(
-                    resolveRefFromUrl(ref, resolutionScope).use {
-                        JSONObject(JSONTokener(it.reader(Charsets.UTF_8)))
-                    },
-                    resolutionScope
-            )
+            val resolvedSchema = resolveRefFromUrl(ref, resolutionScope)
+
+            val resolvedFragment = resolvedSchema.inputStream.use {
+                val fullObject = JSONObject(JSONTokener(it.reader(Charsets.UTF_8)))
+
+                if (!resolvedSchema.fragment.isNullOrBlank()) {
+                    val queryResult: Any? = fullObject.optQuery(resolvedSchema.fragment)
+
+                    if (queryResult !is JSONObject) {
+                        throw java.lang.IllegalArgumentException("Target of fragment ${resolvedSchema.fragment} is not an object")
```

Comment:
Please throw a the kotlin variety of this exception (it's also already used in this class).

Reply:
Done.